src/ngraph/runtime/inference_engine/ie_executable.cpp

```diff
@@ -0,0 +1,143 @@
+
+    for (auto& node : func->get_ops())
+    {
+        if (!opset.contains_op_type(node.get()))
+        {
+            if (node->get_type_info() == op::GetOutputElement::type_info)
+            {
+                // IE currently can handle GetOutuputElement op;
+                continue;
+            }
+            else
+            {
+                std::cout << "UNSUPPORTED OP DETECTED: " << node->get_type_info().name << std::endl;
+                THROW_IE_EXCEPTION << "Detected op not belonging to opset1!";
+            }
+        }
+    }
+
+#ifdef NGRAPH_DEBUG_ENABLE
+    std::cout << "Nodes in test: ";
+    for (auto& node : func->get_ops())
+    {
+        std::cout << node << std::endl;
+    }
+    std::cout << std::endl;
+#endif
+
+    network = InferenceEngine::CNNNetwork(func);
+    device = _device == "INFERENCE_ENGINE" ? "CPU" : _device;
+    set_parameters_and_results(*func);
+}
+
+bool ngraph::runtime::inference_engine::IE_Executable::call(
+    const std::vector<std::shared_ptr<runtime::Tensor>>& outputs,
+    const std::vector<std::shared_ptr<runtime::Tensor>>& inputs)
+{
+    InferenceEngine::Core ie;
+
+    //  Loading model to the plugin (BACKEND_NAME)
+    InferenceEngine::ExecutableNetwork exeNetwork = ie.LoadNetwork(network, device);
+    //  Create infer request
+    InferenceEngine::InferRequest inferRequest = exeNetwork.CreateInferRequest();
+    //  Prepare input and output blobs
+    InferenceEngine::InputsDataMap inputInfo = network.getInputsInfo();
+
+    if (inputInfo.size() != inputs.size())
+    {
+        THROW_IE_EXCEPTION << "Function inputs number differ from number of given inputs";
+    }
+
+    size_t i = 0;
+    for (auto& it : inputInfo)
+    {
+        std::shared_ptr<runtime::inference_engine::IETensorView> tv =
+            std::static_pointer_cast<runtime::inference_engine::IETensorView>(inputs[i]);
+        size_t size = tv->data.size() / sizeof(float);
+        float* orig_data = (float*)tv->data.data();
+        std::vector<float> data(orig_data, orig_data + size);
+        inferRequest.SetBlob(it.first, fill_blob(it.second->getTensorDesc().getDims(), data));
+        i++;
+    }
+
+    //  Prepare output blobs
+    std::string output_name = network.getOutputsInfo().begin()->first;
+
+    inferRequest.Infer();
+    InferenceEngine::Blob::Ptr output = inferRequest.GetBlob(output_name);
+
+    InferenceEngine::MemoryBlob::Ptr moutput =
+        InferenceEngine::as<InferenceEngine::MemoryBlob>(output);
+    if (!moutput)
+    {
+        THROW_IE_EXCEPTION << "Cannot get output MemoryBlob in call_with_validate()";
+    }
+
+    auto lm = moutput->rmap();
+    float* output_ptr = lm.as<float*>();
+    // TODO: how to get size without explicit calculation?
+    size_t size = 1;
+    for (const auto& dim : output->getTensorDesc().getDims())
+    {
+        size *= dim;
+    }
+    //  Vector initialization from pointer
+    std::vector<float> result(output_ptr, output_ptr + size);
```

Comment:
There is no need to make a copy of the data here, just use output_ptr in the write call below.